### ISSUE TITLE
Should I keep decreasing the learning rate?

### ISSUE THREAD

**Issue by txemaheredia** [broadinstitute/CellBender#360]
Hi,

I ran CellBender on my dataset using these params:

```
cellbender remove-background \
                 --cuda \
                 --input ${sample}/raw_feature_bc_matrix.h5 \
                 --output ${sample}/cellbender_output.h5 \
                 --expected-cells 10000 \
                 --total-droplets-included 30000 \
                 --exclude-feature-types "Antibody Capture" \
                 --fpr 0.01 \
                 --epochs 150
```

The html report showed this ELBO plot:

![cb_elbo_1](https://github.com/broadinstitute/CellBender/assets/1544396/d781051c-63ab-40ad-8041-03d5459640e5)

And it issued this warning:

> **Automated assessment --------**
> 
> * WARNING: The training ELBO deviates quite a bit from the max value during the second half of training.  
> * We typically expect to see the training ELBO increase almost monotonically. This curve seems to have a concerted period of motion in the wrong direction near epoch 56. If this is early in training, this is probably okay.  
> * We hope to see the test ELBO follow the training ELBO, increasing almost monotonically (though there will be deviations, and that is expected). There may be a large gap, and that is okay. However, this curve ends with a low test ELBO compared to the max test ELBO value during training. The output could be suboptimal.  
>
> **Summary:**
> 
> This is unusual behavior, and a reduced --learning-rate is indicated. Re-run with half the current learning rate and compare the results.

I followed the suggestion and halved the learning rate, re-running cellbender with `--learning-rate 0.00005`. The resulting ELBO plot looked like this:

![cb_elbo_2](https://github.com/broadinstitute/CellBender/assets/1544396/66062279-c77e-4697-b64f-9b5559a0732d)

> **Automated assessment --------**
> 
> * The training ELBO deviates quite a bit from the max value at the last epoch.  
> * We typically expect to see the training ELBO increase almost monotonically. This curve seems to have a concerted period of motion in the wrong direction near epoch 76. If this is early in training, this is probably okay.  
> * We hope to see the test ELBO follow the training ELBO, increasing almost monotonically (though there will be deviations, and that is expected). There may be a large gap, and that is okay. However, this curve ends with a low test ELBO compared to the max test ELBO value during training. The output could be suboptimal.  
>
> **Summary:**
> 
> This is slightly unusual behavior, and a reduced --learning-rate might be indicated. Consider re-running with half the current learning rate to compare the results.

Should I keep halving the learning rate? Or I'd be better using the default learning rate but using more epochs?

**Comment by robertzeibich** [comment]
Hi @txemaheredia, what did you do at the end?

**Comment by txemaheredia** [comment]
Hi @robertzeibich, 

Apologies for the extremely late reply.

I've checked back my scripts, and I re-run CellBender multiple times using:

- Default parameters (--learning-rate 0.0001 --epochs 150)
- Default Learning Rate + 300 epoch (--learning-rate 0.0001 --epochs 300)
- Half LR ( --learning-rate 0.00005)
- Half LR +  300 epoch
- 1/4 LR + 150 epoch (--learning-rate 0.000025)
- 1/10 LR + 150 epoch (--learning-rate 0.00001)

I did the whole thing for 2 samples I had. After some amount of manual evaluation whose details I don't remember, for one of the samples I used "halfLR_300epoch" and for the other "quartLR_150epoch" as the input for downstream analyses.

Again, apologies for the late reply. 

I hope this serves at least as documentation for future readers.
